### PR TITLE
alpine for docker image

### PR DESCRIPTION
**Pull request by DariuszPorowski**
Do not see any issue for this, but somehow it may be related to https://github.com/lycheeverse/lychee/issues/59 :)

- docker: add alpine flavor for CI docker image
  - reduces image from ~37 MB (debian) to ~12 MB on the registry side (compressed)
  - reduces image from ~104 MB (debian) to ~30 MB on the local disk (uncompressed)
- security: run lychee as non-root user for alpine
- docs: add example about running lychee on Windows PowerShell
- docs: add `--rm` flag to docker run command to remove container after the run (self-cleanup)
- docs: add `-w /input` flag to docker run command to set working directory to /input
  - better UX when running lychee from docker locally (feeling like cli)
- ci: tag/build/push for alpine

bonus:
- docker related github-actions: bump (because of https://github.com/lycheeverse/lychee/actions/runs/5034258304/jobs/9028968846#step:3:54)
- dependabot: auto-bump gh actions
- GitHub App usage for GITHUB_TOKEN
- minor typos fix

### CODE REVIEW

**Comment by mre** [comment]
Thanks @DariuszPorowski,
that's a great achievement. I personally use the Debian image as a base for custom additions, so I'd prefer if the main image stayed Debian for that reason. However, what do you think about renaming the image, such that we could support both Debian and Alpine? E.g. how about `lycheeverse/lychee:0.13-alpine` for example? That's also how PHP does it. ([Reference](https://hub.docker.com/_/php/tags))

**Comment by DariuszPorowski** [comment]
Love the idea, will update PR probably later today or tomorrow :)

**Comment by DariuszPorowski** [comment]
@mre tried to figure out your current docker tags

current:

latest -> debian
master -> debian
0.13 -> debian
0.13.0 -> debian
sha-1234567 -> debian

proposal for alpine:

latest-alpine -> alpine
master-alpine -> alpine
0.13-alpine -> alpine
0.13.0-alpine -> alpine
sha-1234567-alpine -> alpine

Am I missed something? :)

**Comment by mre** [comment]
Yes, that's perfect. 👏 Don't think you missed any tag.

**Comment by DariuszPorowski** [comment]
@mre so, I think Pr is ready :)

File: README.md [hunk 1]
```diff
@@ -188,28 +188,49 @@ but in case you need dedicated support for a new file format, please consider cr
...
 Here's how to mount a local directory into the container and check some input
-with lychee. The `--init` parameter is passed so that lychee can be stopped
-from the terminal. We also pass `-it` to start an interactive terminal, which
-is required to show the progress bar.
+with lychee.
+
+- The `--init` parameter is passed so that lychee can be stopped from the terminal.
+- We also pass `-it` to start an interactive terminal, which is required to show the progress bar.
+- The `--rm` removes not used anymore container from the host after the run (self-cleanup).
+- The `-w /input` points to /input as the default workspace
```
Comment (mre) [inline README.md:196]:
```suggestion
- The `-w /input` points to `/input` as the default workspace
```

File: README.md [hunk 2]
```diff
@@ -188,28 +188,49 @@ but in case you need dedicated support for a new file format, please consider cr
...
-is required to show the progress bar.
+with lychee.
+
+- The `--init` parameter is passed so that lychee can be stopped from the terminal.
+- We also pass `-it` to start an interactive terminal, which is required to show the progress bar.
+- The `--rm` removes not used anymore container from the host after the run (self-cleanup).
+- The `-w /input` points to `/input` as the default workspace
+- The `-v $(pwd):/input` does local volume mounting to the container for lychee access.
+
+> By default debian-based docker image is used. If you want to run alpine-based image, use the `latest-alpine` tag.
```
Comment (mre) [inline README.md:199]:
```suggestion
> By default a Debian-based docker image is used. If you want to run an Alpine-based image, use the `latest-alpine` tag.
```

File: README.md [hunk 3]
```diff
@@ -188,28 +188,49 @@ but in case you need dedicated support for a new file format, please consider cr
...
-is required to show the progress bar.
+with lychee.
+
+- The `--init` parameter is passed so that lychee can be stopped from the terminal.
+- We also pass `-it` to start an interactive terminal, which is required to show the progress bar.
+- The `--rm` removes not used anymore container from the host after the run (self-cleanup).
+- The `-w /input` points to `/input` as the default workspace
+- The `-v $(pwd):/input` does local volume mounting to the container for lychee access.
+
+> By default a Debian-based docker image is used. If you want to run an Alpine-based image, use the `latest-alpine` tag.
```
Comment (mre) [inline README.md:199]:
```suggestion
> By default a Debian-based Docker image is used. If you want to run an Alpine-based image, use the `latest-alpine` tag.
```

File: README.md [hunk 4]
```diff
@@ -188,28 +188,49 @@ but in case you need dedicated support for a new file format, please consider cr
...
-set an environment variable with your Github token like so `GITHUB_TOKEN=xxxx`,
+set an environment variable with your GitHub token like so `GITHUB_TOKEN=xxxx`,
 or use the `--github-token` CLI option. It can also be set in the config file.
 [Here is an example config file][config file].
 
-The token can be generated in your
-[GitHub account settings page](https://github.com/settings/tokens). A personal
-token with no extra permissions is enough to be able to check public repos links.
+The token can be generated on your [GitHub account settings page](https://github.com/settings/tokens).
+A personal access token with no extra permissions is enough to be able to check public repos links.
```
Comment (mre) [inline README.md:222]:
```suggestion
A personal access token with no extra permissions is enough to be able to check public repo links.
```

File: README.md [hunk 5]
```diff
@@ -487,7 +508,7 @@ Lychee makes heavy use of async code to be resource-friendly while still being
 performant. Async code can be difficult to troubleshoot with most tools,
 however. Therefore we provide experimental support for
 [tokio-console](https://github.com/tokio-rs/console). It provides a top(1)-like
-overview for async tasks!
+an overview for async tasks!
```
Comment (mre) [inline README.md:511]:
```suggestion
a overview for async tasks!
```

[the remark this answers is not in the document]
Comment (mre) [reply README.md:511]: This is no typo, actually. The indefinite article, "an", is not needed.

(*In this sentence, "top(1)-like" is an adjective modifying "overview". No need to put an article between an adjective and a noun.*)

**Comment by mre** [comment]
Awesome. Just fixed a few typos (and added a comment for reference, in case you want to have a look). Other than that looks great! ✨ Merging!

**Comment by DariuszPorowski** [comment]
@mre btw... just saw `latest` tag is bind to master branch, not sure it's the best approach ;) IMHO latest should point to the latest production release, so currently 0.13.0

**Comment by mre** [comment]
🤔 you're probably right. Up for another pull request? 😅

**Comment by DariuszPorowski** [comment]
@mre sure, but not this week, probably closer to end of the following one.

**Comment by mre** [comment]
Of course, no rush. 👍

**Pull request merged**